src/metrics.jl

```diff
@@ -140,6 +148,22 @@ see [`Euclidean`](@ref).
 """
 SqEuclidean() = SqEuclidean(0)
 
+"""
+    PeriodicEuclidean(L)
+Create a Euclidean metric on a rectangular periodic domain (i.e., a torus or
+a cylinder). Periods per dimension are contained in the vector `L`.
+For dimensions without periodicity put `Inf` in the respective component.
+
+# Example
+```
+julia> x, y, L = [0.0, 0.0], [0.7, 0.0], (0.5, Inf)
+([0.0, 0.0], [0.7, 0.0], (0.5, Inf))
+
+julia> evaluate(PeriodicEuclidean(L),x,y)
+```
+"""
+PeriodicEuclidean() = Euclidean()
```

Comment:
It feels weird to have a constructor return an object of a different type. Better return a `PeriodicEuclidean` object, but define specialized methods for this special case (where `N==0`).

Also probably better define this at the same time as the `struct` (I'm not sure why `Euclidean` is defined here).